sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample1_SubmitSparkJobAsync.md

```diff
@@ -0,0 +1,68 @@
+# Create, Run and Cancel Synapse Spark jobs
+
+This sample demonstrates basic asynchronous operations with two core classes in this library: `SparkBatchClient` and `SparkBatchJob`. `SparkBatchClient` is used to interact with Spark jobs running on Azure Synapse - each method call sends a request to the service's REST API. `SparkBatchJob` is an entity that represents a batched Spark job within Synapse. The sample walks through the basics of creating, running, and canceling job requests. To get started, you'll need a connection endpoint to Azure Synapse. See the [README](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Spark/README.md) for links and instructions.
+
+## Create Spark batch client
+
+To interact with Spark jobs running on Azure Synapse, you need to instantiate a `SparkBatchClient`. It requires an endpoint URL and a `TokenCredential`.
+
+```C# Snippet:CreateSparkBatchClientAsync
+SparkBatchClient client = new SparkBatchClient(new Uri(endpoint), sparkPoolName, new DefaultAzureCredential());
+```
+
+## Submitting Spark jobs
+
+To submit a Spark job, first create a `SparkBatchJob`, passing in an instance of `SparkBatchJobOptions` describing the job's parameters. Calling `StartCreateSparkBatchJobAsync` with that job will submit it to Synapse.
+
+```C# Snippet:SubmitSparkBatchJobAsync
+string name = $"batch-{Guid.NewGuid()}";
+string file = string.Format("abfss://{0}@{1}.dfs.core.windows.net/samples/net/wordcount/wordcount.zip", fileSystem, storageAccount);
+SparkBatchJobOptions request = new SparkBatchJobOptions(name, file)
+{
+    ClassName = "WordCount",
+    Arguments =
+    {
+        string.Format("abfss://{0}@{1}.dfs.core.windows.net/samples/net/wordcount/shakespeare.txt", fileSystem, storageAccount),
+        string.Format("abfss://{0}@{1}.dfs.core.windows.net/samples/net/wordcount/result/", fileSystem, storageAccount),
+    },
+    DriverMemory = "28g",
+    DriverCores = 4,
+    ExecutorMemory = "28g",
+    ExecutorCores = 4,
+    ExecutorCount = 2
+};
+
+SparkBatchOperation createOperation = await client.StartCreateSparkBatchJobAsync(request);
+SparkBatchJob jobCreated = await createOperation.WaitForCompletionAsync();
+```
+
+## Retrieve a Spark job
+
+To retrieve the details of a Spark job call `StartGetSparkBatchJobAsync`, passing in the Spark job ID.
+
+```C# Snippet:GetSparkBatchJobAsync
+SparkBatchOperation getOperation = await client.StartGetSparkBatchJobAsync (jobCreated.Id);
+SparkBatchJob retrievedJob = await getOperation.WaitForCompletionAsync();
+
+Debug.WriteLine($"Job is returned with name {retrievedJob.Name} and state {retrievedJob.State}");
+```
+
+## List Spark jobs
+
+To enumerate all Spark jobs in the Synapse workspace call `GetSparkBatchJobs`.
+
+```C# Snippet:ListSparkBatchJobsAsync
```

Comment:
name of the snippet suggests the method is Async.
Which also makes me curious, is it intentional that the sample.md is showing a combination of async and sync approach?

Reply:
Yeah, I wasn't sure here. I wanted to have all of the snippets from Sample1_SubmitSparkJobAsync.md to come from Sample1_SubmitSparkJobAsync.cs but there is no difference here between Sample1_SubmitSparkJobAsync.cs and Sample1_SubmitSparkJob.cs on this set.

Also, there is no async variant of this API.

Suggestions?

Reply:
Does that mean that there is a sync version of all these operations? if so, I will suggest making your sample all sync and put Notes or links at the bottom suggesting that there are also async ways to do it. 
We don't fully do that in [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/samples/Sample1_DetectLanguage.md), but u can get an idea.

Reply:
ohhh I see u have the sync version already. Will it be too bad to delete the async.md and just link to the async class in the other one?